src/common/front-end-routing/route-injection-token.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import type { LensRendererExtension } from "../../extensions/lens-renderer-extension";
+
+export const routeInjectionToken = getInjectionToken<Route<unknown>>({
+  id: "route-injection-token",
+});
+
+export interface Route<TParameter = void> {
+  path: string;
+  clusterFrame: boolean;
+  isEnabled: IComputedValue<boolean>;
```

Comment:
How about we type this as `boolean | IComputedValue<boolean>` to avoid the `computed(() => true)`?

Reply:
Hmm. A conditional type here would come with non-insignificant cost in complexity, as it introduces an alternate way of using `isEnabled`. There is no actual value in minimal implementation of just `true` over `computed(() => true)`. Instead, having to use value of `isEnabled` somewhere without certainty of type is of negative value.

Reply:
The other option could be for this to be optional and if not provided it defaults to `computed(() => true)`

Reply:
So the cost is that caller needs to call `.get()` for standalone computed value, and since there are multiple uses of this `isEnabled` we go with enforcing the computed type.